### PR TITLE
feature/boxed_table_output_for_load_report

### LINKED ISSUE #139: Boxed table output for load-report
**Motivation:**
A text-format load report already exists (`cli/format.py:format_text`) and shows
field origins with masking. The only missing piece is a "boxed" table renderer
(`+------+----------+-------+`) that is easier to scan at a glance — purely cosmetic.

**Proposal:**
Add a `format_table(report, ...)` function in `cli/format.py` (or a `--format=table`
flag for `dature inspect`) that renders a compact ASCII table:
```
  +------------------+----------+--------------+
  | key              | source   | value        |
  +------------------+----------+--------------+
  | database.host    | app.yml  | db.prod.com  |
  | database.pass    | vault    | ****         |
  +------------------+----------+--------------+
```
Masking is already implemented; this is presentation only.

**Acceptance criteria:**
- Tabular output with `key`, `source`, `value` columns; secrets masked.
- Existing `text` and `json` formats unchanged.
- Tests asserting exact output (as per project convention — full `==`, not `in`).
- `changes/` fragment.

**References:** hoplite report mode.

### PR DESCRIPTION
**Pull request by nyashkin**
Closes https://github.com/reagento/dature/issues/139

### CODE REVIEW

File: src/dature/cli/format.py [hunk 1]
```diff
@@ -75,3 +78,58 @@ def format_text(report: LoadReport, *, field: str | None = None) -> str:
...
+    return "+" + "+".join("-" * (width + 2) for width in widths) + "+"
+
+
+def _table_row(values: tuple[str, ...], widths: tuple[int, ...]) -> str:
+    return "|" + "|".join(f" {value:<{width}} " for value, width in zip(values, widths, strict=True)) + "|"
+
+
+def _table_source_label(origin: FieldOrigin) -> str:
+    source = origin.source_file or origin.source_loader_type
+    path = PureWindowsPath(source)
```
Comment (Niccolum) [inline src/dature/cli/format.py]: Problem: PureWindowsPath is used to extract the basename, but it treats backslash as a path separator even on POSIX systems, where a backslash can be a legal character inside a filename. A file literally named "config\special.yml" would get truncated incorrectly.

Fix: use a single, OS-independent split (e.g. split on both "/" and "" explicitly, or just use PurePath and drop the Windows-specific branch), and add a one-line comment explaining why both separators need to be handled.

File: src/dature/cli/format.py [hunk 2]
```diff
@@ -75,3 +78,58 @@ def format_text(report: LoadReport, *, field: str | None = None) -> str:
...
+    if path.name != source:
+        return path.name
+    return PurePath(source).name
+
+
+def _table_value_repr(value: JSONValue) -> str:
+    if isinstance(value, str):
+        if value == config.masking.mask:
```
Comment (Niccolum) [inline src/dature/cli/format.py]: Problem: this checks value == config.masking.mask to decide whether to print "", but the real masked value produced by mask_value() in masking.py is prefix + mask + suffix, not just the bare mask string, whenever visible_prefix or visible_suffix is non-zero. So this check silently fails for that config and the raw mask marker (e.g. "<REDACTED>") leaks into the table instead of being replaced by "". Alsoformat_text does none of this — it just prints the already-masked value as-is, since masking already happened upstream in report.py.

Fix: remove this branch entirely and just return value unchanged, matching what format_text does. Don't try to re-detect masking at the presentation layer.

File: src/dature/cli/format.py [hunk 3]
```diff
@@ -75,3 +78,58 @@ def format_text(report: LoadReport, *, field: str | None = None) -> str:
...
+
+def format_table(report: LoadReport, *, field: str | None = None) -> str:
+    headers = ("key", "source", "value")
+    rows: list[tuple[str, str, str]] = []
+    for origin in _filter_origins(report, field):
+        source = _table_source_label(origin)
+        rows.extend(
+            (key, source, _table_value_repr(value)) for key, value in _flatten_table_values(origin.key, origin.value)
+        )
+    widths = tuple(max(len(headers[i]), *(len(row[i]) for row in rows)) for i in range(3))
```
Comment (Niccolum) [inline src/dature/cli/format.py]: Problem: this line crashes with TypeError: 'int' object is not iterable whenever rows is empty — e.g. when --field matches nothing, or the report has no field origins. I reproduced this locally.

Fix: change to max([len(headers[i]), *(len(row[i]) for row in rows)]) (wrap in a list so max() gets an iterable even with zero rows), and add a test for the empty-rows case.

File: src/dature/cli/format.py [hunk 4]
```diff
@@ -75,3 +78,58 @@ def format_text(report: LoadReport, *, field: str | None = None) -> str:
...
+
+def _table_source_label(origin: FieldOrigin) -> str:
+    source = origin.source_file or origin.source_loader_type
+    path = PureWindowsPath(source)
+    if path.name != source:
+        return path.name
+    return PurePath(source).name
+
+
+def _table_value_repr(value: JSONValue) -> str:
```
Comment (Niccolum) [inline src/dature/cli/format.py:95]: Problem: non-string values go through json.dumps, which escapes newlines and quotes, but string values are returned raw with no escaping. A string value containing a newline will break the table's row/border alignment, since the table format assumes one line per row.

Fix: escape or sanitize string values the same way non-string values are handled (e.g. run them through repr() or replace "\n" with a visible placeholder) before putting them in a table cell.

File: src/dature/cli/format.py [hunk 5]
```diff
@@ -75,3 +78,58 @@ def format_text(report: LoadReport, *, field: str | None = None) -> str:
...
+        return value
+    return json.dumps(value, ensure_ascii=False, default=str)
+
+
+def _flatten_table_values(key: str, value: JSONValue) -> tuple[tuple[str, JSONValue], ...]:
+    if not isinstance(value, dict) or not value:
+        return ((key, value),)
+
+    rows: list[tuple[str, JSONValue]] = []
+    for child_key, child_value in sorted(value.items()):
```
Comment (Niccolum) [inline src/dature/cli/format.py]: Problem: nested dict keys are sorted alphabetically here, but format_text and format_json preserve the original field order. That means the same report can show fields in a different order depending on --format, which is confusing and inconsistent.

Fix: preserve insertion order here too (drop the sorted() call)

File: src/dature/cli/format.py [hunk 6]
```diff
@@ -1,9 +1,12 @@
 import json
 from dataclasses import asdict
+from pathlib import PurePath, PureWindowsPath
 from typing import Any
 
+from dature.config import config
```
Comment (Niccolum) [inline src/dature/cli/format.py]: remove according format.py:101 comment

File: tests/cli/test_inspect.py
Comment (Niccolum) [inline tests/cli/test_inspect.py:1]: new tests only cover a flat 2-field config and one masked secret. The following cases aren't tested at all, and some of them hit real bugs:

  - nested dict values — this is the only non-trivial new logic (_flatten_table_values) and it has zero direct test coverage

  - --field that matches no origins, or an empty report — currently crashes (see format.py:125 comment)

  - --field combined with --format table together

  - list-valued fields — not flattened by _flatten_table_values, falls through to json.dumps; needs a test to pin down that this is the intended behavior

  - multiple --source entries merged, with --format table — verify the source column shows the correct file per origin

  - a value containing a newline or special characters — to document the current row-alignment breakage (see format.py:99-104 comment)

  - a None value

  - a row where every value is shorter than the header word "value" (5 chars), to exercise the branch where the header width wins over the row width

File: docs/examples/basic/cli/cli_inspect_help.stdout [hunk 7]
```diff
@@ -5,7 +5,7 @@ usage: dature inspect [-h] --schema MODULE:ATTR --source SPEC
                       [--expand-env-vars {disabled,default,empty,strict}]
```
Comment (Niccolum) [inline docs/examples/basic/cli/cli_inspect_help.stdout:5]: add example .sh and .stdout for table (and json, what i forgot). See docs/examples/basic/cli/cli_quickstart_inspect_text.sh for example. And add it to docs

File: src/dature/cli/format.py [hunk 8]
```diff
@@ -75,3 +78,58 @@ def format_text(report: LoadReport, *, field: str | None = None) -> str:
...
+
+def _table_border(widths: tuple[int, ...]) -> str:
+    return "+" + "+".join("-" * (width + 2) for width in widths) + "+"
+
+
+def _table_row(values: tuple[str, ...], widths: tuple[int, ...]) -> str:
+    return "|" + "|".join(f" {value:<{width}} " for value, width in zip(values, widths, strict=True)) + "|"
+
+
+def _table_source_label(origin: FieldOrigin) -> str:
```
Comment (Niccolum) [inline src/dature/cli/format.py:89]: _table_source_label shows only the filename, while text/json show the full source path. Please - use the full source path

File: tests/cli/test_inspect.py
Comment (nyashkin) [reply tests/cli/test_inspect.py:1]: thnx, agreed. I've added coverage for all of these edge cases, including nested fields, empty/missing origins, `--field` + table, lists, multiple sources, newlines/special characters, `None`, and header-driven column width.

File: src/dature/cli/format.py (hunk 6 above)
Comment (nyashkin) [reply src/dature/cli/format.py]: done

File: src/dature/cli/format.py (hunk 5 above)
Comment (nyashkin) [reply src/dature/cli/format.py]: Fixed - removed `sorted()` so nested fields preserve their insertion order consistently across formats.

File: src/dature/cli/format.py (hunk 4 above)
Comment (nyashkin) [reply src/dature/cli/format.py:95]: Fixed - string values are now escaped with `repr()`, so newlines and special characters can't break the table layout.

File: src/dature/cli/format.py (hunk 3 above)
Comment (nyashkin) [reply src/dature/cli/format.py]: Fixed. Added an E2E test covering empty `rows`.

File: src/dature/cli/format.py (hunk 2 above)
Comment (nyashkin) [reply src/dature/cli/format.py]: Fixed. Removed the presentation-layer masking check and now use the already-masked value unchanged, consistent with `format_text`.

File: src/dature/cli/format.py (hunk 8 above)
Comment (nyashkin) [reply src/dature/cli/format.py:89]: Fixed - table output now uses the full source path

File: src/dature/cli/format.py (hunk 1 above)
Comment (nyashkin) [reply src/dature/cli/format.py]: Fixed

File: docs/examples/basic/cli/cli_inspect_help.stdout (hunk 7 above)
Comment (nyashkin) [reply docs/examples/basic/cli/cli_inspect_help.stdout:5]: thnx, I've added an example of the table. JSON output already exists in cli.md. What exactly did you mean?

Comment (Niccolum) [reply docs/examples/basic/cli/cli_inspect_help.stdout:5]: Oh, really, json already exists. Thanks!

**Review by Niccolum — APPROVED** [review]

**Comment by Niccolum** [comment]
I'll merge this manually from my side this time, but I highly recommend setting up commit signatures for the future. Right now, it's blocked because of the "Commits must have verified signatures" rule. 😊
